### PR TITLE
Adding new filter for ObsPack data retrieval from ICOS Carbon Portal

### LINKED ISSUE #788: Retrieving European Obspack data from ICOS CP - specifying files
### What is your issue?

There have been a couple of changes on the ICOS CP that have broken the capability of openghg.retrieve.icos.retrieve_atmospheric to correctly import European Obspack data:

1) The .nc files were previously skipped through the fix in #630 . This searches dobj.meta["references"]["doi"]["titles"][0]["title"].lower() for "globalview" and skips when found, however the new .nc files no longer have "globalview" in this attribute. So a new solution is required to skip them.

2) The 2022 European Obspack data are still returned by the search, even though these data are now deprecated and superseded by the 2023 versions. So we need to be able to filter by version.

For example - try running:
cp_data = retrieve_atmospheric(site="TAC",
                               force_retrieval=True,
                               data_level=2,
                               dataset_source="European ObsPack",
                               update_mismatch="from_source",
                               species="co2",
                               inlet="54m")

This tries to retrieve three records (and fails on the third of these):
https://meta.icos-cp.eu/objects/MWN6tAJF49pvqY3B09KBkhJa
https://meta.icos-cp.eu/objects/1etvTvRIvxOIQ9Wut4RpRFUE
https://meta.icos-cp.eu/objects/MtJYBO6tu3qHn62u2oSIDIEF

We only want it to return the second of these records (although it might be useful to retain the ability to retrieve the first of these records too)

### PR DESCRIPTION
**Pull request by gareth-j**
* **Summary of changes** (Bug fix, feature, docs update, ...)

Quick fix for ICOS data retrieval.

> [!NOTE]
> Tests aren't passing at the moment and I haven't got time to fix them at the moment, will keep this as a draft for now

* **Please check if the PR fulfills these requirements**

- [x] Closes #788
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature

### CODE REVIEW

File: tests/retrieve/icos/test_retrieve_icos.py [hunk 1]
```diff
@@ -148,17 +148,37 @@ def test_icos_retrieve_skips_obspack_globalview(mocker, caplog):
 
     # Check attributes within stored Dataset contain extra keys
     attr1_expected_additional = {
-        "measurement_type": "co2 mixing ratio (dry mole fraction)",
+        "species": "co2",
+        "instrument": "FTIR",
+        "instrument_data": ["FTIR", "http://meta.icos-cp.eu/resources/instruments/ATC_505"],
+        "site": "WAO",
+        "measurement_type": "co2 (dry air mole fraction)",
+        "units": "µmol mol-1",
```
Comment (rt17603) [inline tests/retrieve/icos/test_retrieve_icos.py:156]: We're planning to remove "units" from the global attributes at some point in relation to #541 (see also branch rm_units_metadata). We do also need to make sure the units details are converted and stored against the individual species.

Should we remove this in this PR? Or did we want to include it PR but to reference this in the linked issue so we make sure to cover this when fixing it? Thoughts?

**Review by rt17603 — COMMENTED** [review]
Solution looks reasonable to me but I've also just asked Joe in case this may have more wide reaching impacts than we're anticipating. Also added a small comment on whether to include "units" as a global attribute.

File: tests/retrieve/icos/test_retrieve_icos.py (hunk 1 above)
Comment (gareth-j) [reply tests/retrieve/icos/test_retrieve_icos.py:156]: Yeah I can remove that now.

Comment (gareth-j) [reply tests/retrieve/icos/test_retrieve_icos.py:156]: OK let's leave the units attribute as is for now.

**Review by joe-pitt — COMMENTED** [review]
This is working nicely for me at the moment! I guess there's always a risk going forward that the labels associated with the files we want to exclude change again, but we can deal with that if/when it happens...

**Review by rt17603 — APPROVED** [review]
Ok, looks like a useful fix for now. A few small requests for the PR description:
 - Can this be updated to remove the note that the tests aren't passing (I'm assuming they now are?)
 - Can this be updated to summarise change made e.g. updating ICOS retrieve function to ignore all names containing "ObsPack".

**Pull request merged**